src/parser/type_checking.c

```diff
@@ -13263,25 +13247,15 @@ pt_eval_function_type (PARSER_CONTEXT * parser, PT_NODE * node)
 
     case F_JSON_ARRAY:
       {
-	PT_TYPE_ENUM supported_types[] =
-	  { PT_TYPE_CHAR, PT_TYPE_INTEGER, PT_TYPE_DOUBLE, PT_TYPE_JSON, PT_TYPE_NUMERIC, PT_TYPE_NULL, PT_TYPE_MAYBE };
 	PT_TYPE_ENUM unsupported_type;
-	int len = sizeof (supported_types) / sizeof (supported_types[0]);
-	int i, found_supported_type = 0;
+	bool found_supported_type;
 
 	PT_NODE *arg = arg_list;
 
 	while (arg)
 	  {
-	    found_supported_type = 0;
-	    for (i = 0; i < len; i++)
-	      {
-		if (supported_types[i] == arg->type_enum)
-		  {
-		    found_supported_type = 1;
-		    break;
-		  }
-	      }
+	    found_supported_type = pt_is_json_value_type (arg->type_enum);
+
 	    if (!found_supported_type)
```

Comment:
use directly:
```c
  if (!pt_is_json_value_type (arg->type_enum))
```

Reply:
found_supported_type is also used outside of the while loop

Reply:
ok, you're right. but please rename is as is_supported, the name is still confusing.